app/js/controllers/main.js

```diff
@@ -35,11 +53,82 @@ export default class MainController extends Controller {
     };
   }
 
+  _lazyLoadModules() {
+    return new Promise((resolve) => {
+      if (this._loadedModules) {
+        resolve();
+        return;
+      }
+
+      var editView = new EditView();
+      var actionMenuView = new ActionMenuView();
+      var settingsView = new SettingsView();
+      var viewSourceView = new ViewSourceView();
+      var appendChildView = new AppendChildView();
+      var moveView = new MoveView();
+      var mainView = new MainView({
+        editView: editView,
+        actionMenuView: actionMenuView,
+        settingsView: settingsView,
+        viewSourceView: viewSourceView,
+        appendChildView: appendChildView,
+        moveView: moveView
+      });
+
+      var editController = new EditController({
+        view: editView
+      });
+
+      var viewSourceController = new ViewSourceController({
+        view: viewSourceView
+      });
+
+      var appendChildController = new AppendChildController({
+        view: appendChildView
+      });
+
+      var moveController = new MoveController({
+        view: moveView
+      });
+
+      var actionMenuController = new ActionMenuController({
+        view: actionMenuView,
+        editController: editController,
+        viewSourceController: viewSourceController,
+        appendChildController: appendChildController,
+        moveController: moveController
+      });
+
+      var settingsController = new SettingsController({
+        view: settingsView
+      });
+
+      this.view = mainView;
+      mainView.init(this);
+
+      this.actionMenuController = actionMenuController;
+      this.settingsController = settingsController;
+
+      editController.mainController = this;
+      viewSourceController.mainController = this;
+      actionMenuController.mainController = this;
+      settingsController.mainController = this;
+      appendChildController.mainController = this;
+      moveController.mainController = this;
```

Comment:
Why not just use ```window.mainController``` instead? Again, we're moving code around, so I'm not going to make this an action either, but it's a good potential refactor.